tools/clang/unittests/HLSL/ExecutionTest.cpp

```diff
@@ -8478,7 +8478,7 @@ void VerifyAtomicResults(const BYTE *uResults, const BYTE *sResults,
   // This is interpretted as -maxIndex and will be the lowest
   // The maxIndex will be unaltered and interpretted as the highest.
   LogCommentFmt(L"Verifying %d-bit integer atomic smin", bitSize);
-  VERIFY_IS_TRUE(AtomicResultMatches(sResults + stride*SMIN_IDX, SHIFT(-((int)maxIdx-1), shBits), byteSize)); // SMin
+  VERIFY_IS_TRUE(AtomicResultMatches(sResults + stride*SMIN_IDX, SHIFT(-((int64_t)maxIdx-1), shBits), byteSize)); // SMin
```

Comment:
I think there's a fundamental problem using size_t for maxIdx and expecting that to be 64-bits.  We need to use a uint64_t.  Then should we be using `~` instead of `-` here?  Then no need for signed values, right?

Reply:
maxIdx never exceeds the maximum value expressed by a 32-bit integer. SHIFT converts it to a 64-bit value as it copies the lower 32 bits into the upper 32 bits. The corresponding parameter to AtomicResultMatches is uint64_t.  As you say, there is nothing wrong in this location. If maxIdx were uint64_t, this cast would still be necessary. The warning as error in question is regarding applying a negation operator to an unsigned value